Sprint 6/step 1

artem-benda opened the pull request:
- unit tests
- cypress e2e tests
- updated README with deployments info, running tests instructions

artem-benda committed:
added auth reducer tests

added
- testing libs
- auth reducer tests

artem-benda committed:
added reducers tests

added reducers tests

artem-benda committed:
added cypress e2e tests

added
- cypress e2e tests

updated
- README

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* проект задеплоен и отлично работает
* все 48 юнит-тестов проходят успешно
* cypress-тест проходит успешно


но есть некоторые недочеты:

* почему-то в пулл-реквесте в измененных файлах нет кода юнит-тестов, чтобы я мог там оставить комментарии. Придется тут просто описать:
 * во всех юнит-тестах дублируется огромное количество одинаковых данных в объектах. Чтобы не копировать их из теста в тест, нужно вынести их в константы и использовать в коде.


Пример из `auth.text.js` у Вас. Создаем `initialState` и используем его во всем код, меняя точечно данные, которые меняются в объекте с помощью `spread`-операции

```js


const initialState = {  // константный объект
    user: null,

    loginRequest: false,
    loginFailed: false,

    logoutRequest: false,
    logoutFailed: false,

    registerRequest: false,
    registerFailed: false,

    sendResetPasswordCodeRequest: false,
    sendResetPasswordCodeSuccess: false,
    sendResetPasswordCodeFailed: false,

    resetPasswordRequest: false,
    resetPasswordSuccess: false,
    resetPasswordFailed: false,

    getUserRequest: false,
    getUserFailed: false,

    editUserRequest: false,
    editUserFailed: false
}

describe('auth reducer', () => {
    it('should return the initial state', () => {
        expect(authReducer(undefined, {})).toEqual(initialState);
    });

    it('should handle LOGIN_REQUEST', () => {
        expect(
            authReducer(initialState, {
            type: types.LOGIN_REQUEST
        })
        ).toEqual({
            ...initialState,  // spread-операция, чтобы не дублировать код
            loginRequest: true,
        });
    });



```

artem-benda committed:
fix по результатам код-ревью

fix по результатам код-ревью

для тестов reducer'ов теперь начальное состояние вынесено в отдельную константу и оно используется во всех тестах

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.


Удачи в Вашей новой профессии.

**Несколько советов для дальнейшего развития:**

* сверстайте несколько небольших проектов для себя, для закрепления знаний, и чтобы можно добавить их в портфолио
* ходите на собеседования: опыт прохождения собеседований не менее важен знаний теории, и Вы поймете, какие знания требуются на рынке, и где Вы проседаете
* читайте [habr](https://habr.com/ru/) и слушаете подкаксты: они очень хорошо расширяют кругозор и позволяют узнать, что происходит в отрасли:

https://soundcloud.com/everyonecan

https://soundcloud.com/podlodka

https://soundcloud.com/devschacht

https://soundcloud.com/begebot

https://soundcloud.com/web-standards

https://www.youtube.com/channel/UCZeU17nbVfzczAkJVTay9vw
